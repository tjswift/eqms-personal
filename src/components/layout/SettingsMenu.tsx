import * as React from 'react';
import LayoutContainer from '../../containers/LayoutContainer';
import { Menu, MenuItem, MenuDivider } from '@blueprintjs/core';

//todo: add settings specific menu items and props

interface SettingsProps {
    
}

const SettingsMenu: React.SFC<SettingsProps> = ({  }) => (
    <Menu>
        <MenuItem text="Report Feedback" href="https://nsieng.atlassian.net/rest/collectors/1.0/template/form/b46977fb?os_authType=none#"/>
        <MenuDivider />
        <MenuItem text="Item 2" />
        <MenuItem text="Item 3" />
        <MenuDivider />
        <LayoutContainer>
            {({ theme, setTheme }) => (
                <React.Fragment>
                    <MenuItem text="Sign Out" onClick={() => {}} />
                </React.Fragment>
            )}
        </LayoutContainer>
        
    </Menu>
)

export default SettingsMenu

