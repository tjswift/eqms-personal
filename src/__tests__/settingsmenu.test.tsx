/**
 * Current tests include:
 * it is there, renders, 
 * menu items exist in the correct order
**/
import React from 'react';
import SettingsMenu from '../components/layout/SettingsMenu'
import { shallow, configure, ReactWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<SettingsMenu />', () => {
    
    const menuWrapper = shallow(<SettingsMenu />);
    const menuDiv = menuWrapper.find('Menu');

    it('renders', () => {
        expect(menuDiv).toBeTruthy();
    });

    it('renders all elements in correct order (snapshot)', () => {
        expect(menuWrapper).toMatchSnapshot();
    })


});