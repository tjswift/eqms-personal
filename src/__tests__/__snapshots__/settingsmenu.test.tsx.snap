// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsMenu /> renders all elements in correct order (snapshot) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SettingsMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Blueprint3.MenuItem
          disabled={false}
          multiline={false}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Item 1"
        />,
        <Blueprint3.MenuItem
          disabled={false}
          multiline={false}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Item 2"
        />,
        <Blueprint3.MenuItem
          disabled={false}
          multiline={false}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Item 3"
        />,
        <Blueprint3.MenuDivider />,
        <Connect(LayoutContainer)>
          [Function]
        </Connect(LayoutContainer)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "multiline": false,
          "popoverProps": Object {},
          "shouldDismissPopover": true,
          "text": "Item 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "multiline": false,
          "popoverProps": Object {},
          "shouldDismissPopover": true,
          "text": "Item 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "multiline": false,
          "popoverProps": Object {},
          "shouldDismissPopover": true,
          "text": "Item 3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="Item 1"
          />,
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="Item 2"
          />,
          <Blueprint3.MenuItem
            disabled={false}
            multiline={false}
            popoverProps={Object {}}
            shouldDismissPopover={true}
            text="Item 3"
          />,
          <Blueprint3.MenuDivider />,
          <Connect(LayoutContainer)>
            [Function]
          </Connect(LayoutContainer)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "multiline": false,
            "popoverProps": Object {},
            "shouldDismissPopover": true,
            "text": "Item 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "multiline": false,
            "popoverProps": Object {},
            "shouldDismissPopover": true,
            "text": "Item 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "multiline": false,
            "popoverProps": Object {},
            "shouldDismissPopover": true,
            "text": "Item 3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
